core/src/labels/label.h

```diff
@@ -72,71 +84,94 @@ class Label {
         size_t repeatGroup = 0;
         float repeatDistance = 0;
         float buffer = 0.f;
-
-        // the label hash based on its styling parameters
-        size_t paramHash = 0;
-
+        size_t paramHash = 0; // the label hash based on its styling parameters
         LabelProperty::Anchors anchors;
         bool required = true;
+        bool flat = false;
+        float angle = 0.f;
     };
 
     static const float activation_distance_threshold;
 
-    Label(Transform _transform, glm::vec2 _size, Type _type, Options _options);
+    Label(WorldTransform _transform, glm::vec2 _size, Type _type, Options _options);
 
     virtual ~Label();
 
-    bool update(const glm::mat4& _mvp, const glm::vec2& _screenSize, float _zoomFract, bool _drawAllLabels = false);
+    // Push the pending transforms to the vbo by updating the vertices
+    virtual void updateVertexBuffer() = 0;
+    virtual glm::vec2 center() const;
```

Comment:
Ok this seems accurate with what happens on the dynamic quad mesh, will update for that.